### PR TITLE
Fix broken db update button on homescreen

### LINKED ISSUE #47172: [WC 8.9 Beta] Nothing happens when you click the "Update WooCommerce database" button from the WC Homepage
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Discovered while testing 8.9 Beta.

After upgrading from WC 8.8.3 to WC 8.9 Beta1, the "Update WooCommerce database" notice appears on the WC Homepage. When you click on the "Update WooCommerce database" button there, nothing happens. 

However, when you navigate to a non-WC Admin page, like the Orders page, clicking the "Update WooCommerce database" button correctly initializes the DB update. 

The screenshot below says I upgraded to `8.9.0.40` but please note that this is happening on 8.9 beta as well.

yuPm6k6XzJ
 

### Expected behavior

Not sure what the expected behavior is. Either:
a. The notice does not appear on the WC Homepage, or...
b. The notice would continue to appear on the WC Homepage, but clicking on the "Update WooCommerce database" button should correctly trigger the db update process.

### Actual behavior

Nothing happens when you click on the "Update WooCommerce database" button while you're in the WC Homepage.

### Steps to reproduce

1. Install WC 8.8.3 on a JN site.
2. Install the WooCommerce Beta Tester.
3. Use the WC Beta Tester to upgrade to WC 8.9 Beta.
4. Go to WooCommerce > Home. You would see the "Update WooCommerce database" notice there.
5. Click on the "Update WooCommerce database" button. 
6. The bug: Observe for a few seconds that nothing happens.
7. Now go to any non-WC Admin page like WooCommerce > Settings, or WooCommerce > Orders.
8. You would still see the "Update WooCommerce database" notice as expected.
9. Click on the "Update WooCommerce database" button there.
10. Notice that the db update starts, as expected.

### WordPress Environment

JN site with WP 6.5
WC 8.8.3 that was upgraded to WC 8.9 Beta

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by adrianduffell**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This fixes a bug in the inbox note to update database. The note previously relied on a mechanism to update the db from the current URL the notice is on. However this is not compatible with the home screen, as WooCommerce Admin's will route the URL with JS instead of performing a server load. To get around this, we use the WP dashboard URL (which WCA won't try to route, and implement a redirect back to the original page.

https://github.com/woocommerce/woocommerce/assets/9312929/b452df0e-709a-48ac-a9d3-62ea065a7676

Closes https://github.com/woocommerce/woocommerce/issues/47172 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce > Home in admin
2. See the notice for the update (if it doesn't appear on dev sites, it may require rewinding the db version, e.g `wp option set woocommerce_db_version 8.9.0`
3. Perform the update
5. Confirm the homescreen reloads and shows an updating or thanks message.

To further test around this, reset the database version again and try from other screens.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php [hunk 1]
```diff
@@ -113,8 +113,9 @@ private static function update_needed_notice( $note_id = null ) {
 			add_query_arg(
 				array(
 					'do_update_woocommerce' => 'true',
+					'return_url' => wc_get_current_admin_url() ? wc_get_current_admin_url() : admin_url( 'admin.php?page=wc-settings' ),
```
Comment (chihsuan) [inline plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php]: Fix lint error:

```
. lint:changes:branch:php: FILE: plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php:  116 | WARNING | Array double arrow not aligned correctly; expected 12 space(s) between "'return_url'" and double arrow, but found 1. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
```


```suggestion
					'return_url'            => wc_get_current_admin_url() ? wc_get_current_admin_url() : admin_url( 'admin.php?page=wc-settings' ),
```

File: plugins/woocommerce/includes/class-wc-install.php [hunk 2]
```diff
@@ -424,6 +424,9 @@ public static function install_actions() {
 			self::update();
 			WC_Admin_Notices::add_notice( 'update', true );
 		}
+		if ( ! empty( $_GET['return_url'] ) ) { // WPCS: input var ok.
+			wp_safe_redirect( $_GET['return_url'] ); // WPCS: input var ok.
```
Comment (chihsuan) [inline plugins/woocommerce/includes/class-wc-install.php]: Fix lint error


```
. lint:changes:branch:php: FILE: plugins/woocommerce/includes/class-wc-install.php
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: FOUND 1 ERROR AND 0 WARNINGS AFFECTING 1 LINE
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php:  428 | ERROR | $_GET['return_url'] not unslashed before sanitization. Use wp_unslash() or similar (WordPress.Security.ValidatedSanitizedInput.MissingUnslash)
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
```

```suggestion
			$return_url = esc_url_raw( wp_unslash( $_GET['return_url'] ) );
			wp_safe_redirect( $return_url ); // WPCS: input var ok.
```

**Review by chihsuan — APPROVED** [review]
Tested well. Great idea to use the dashboard URL to avoid the JS routing issue. 👍 

Just left some suggestions to fix linting issues.

**Review by ilyasfoo — APPROVED** [review]
Tests well!

**Pull request merged**